db_stress_tool/db_stress_test_base.cc

```diff
@@ -1163,7 +1205,13 @@ void StressTest::OperateDb(ThreadState* thread) {
         }
 
         if (total_size <= FLAGS_backup_max_size) {
+          // TODO(hx235): enable metadata error injection with
+          // backup/restore after fixing the various issues it surfaces
+          fault_fs_guard->DisableThreadLocalMetadataReadErrorInjection();
+          fault_fs_guard->DisableThreadLocalMetadataWriteErrorInjection();
           Status s = TestBackupRestore(thread, rand_column_families, rand_keys);
+          fault_fs_guard->EnableThreadLocalMetadataWriteErrorInjection();
+          fault_fs_guard->EnableThreadLocalMetadataReadErrorInjection();
           ProcessStatus(shared, "Backup/restore", s);
```

Comment:
Had to do this to focus on surfacing actual code issues. Will add this back once we run out of actual issue to solve. Based on my current experience, adding this back surfaced mostly testing framework issues, some of which were investigated and fixed in this PR.